### PR TITLE
Get reports by asset tags

### PR DESCRIPTION
**Pull request by tspenov**
## Changes

Adding tags to reports - that should be the assets that are in this report.  
Adding api to fetch all reports that contain one of the provided tags.

## Ticket

https://www.notion.so/santiment/Add-asset-tags-for-reports-Filtering-by-tags-assets-f4e7bb1d43924ba796690da5159e6e29

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [x] I have added tests that prove my fix is effective or that my feature works

## Usage

```graphql
   {
      getReportsByTags(tags: ['t1', 't2'])
      {
        url
        name
        tags
      }
    }
```

## Screenshots

image
image
image

### CODE REVIEW

File: priv/repo/structure.sql [hunk 1]
```diff
@@ -1,8 +1,10 @@
 --
```
Comment (IvanIvanoff) [inline priv/repo/structure.sql:1]: What happened here? Why 4327 lines changed? Can you include only the needed changes from this PR here?

File: lib/sanbase/reports/report.ex [hunk 2]
```diff
@@ -65,22 +77,47 @@ defmodule Sanbase.Report do
...
   end
 
-  def get_published_reports(%{plan: %{name: "FREE"}}) do
-    from(r in __MODULE__, where: r.is_published == true and r.is_pro == false)
-    |> Repo.all()
+  # Helpers
+
+  defp normalize_tags(%{"tags" => tags} = attrs) when is_binary(tags) do
+    Map.put(attrs, "tags", String.split(tags, ~r{\s*,\s*}) |> Enum.map(&String.downcase/1))
   end
```
Comment (IvanIvanoff) [inline lib/sanbase/reports/report.ex:90]: Why not just `|> String.split(",")`?

File: lib/sanbase/reports/report.ex [hunk 3]
```diff
@@ -65,22 +77,47 @@ defmodule Sanbase.Report do
     Repo.all(__MODULE__)
   end
 
-  def get_published_reports(nil) do
-    from(r in __MODULE__, where: r.is_published == true and r.is_pro == false)
+  def get_published_reports(subscription) do
+    __MODULE__
+    |> get_published_reports_query(subscription)
```
Comment (IvanIvanoff) [inline lib/sanbase/reports/report.ex:82]: You can use `Subscription.plan_name(subscription)` here so you don't pattern match on subscription internals but just on :pro, :free, etc.

File: lib/sanbase/reports/report.ex (hunk 2 above)
Comment (tspenov) [reply lib/sanbase/reports/report.ex:90]: To remove whitespaces around

File: priv/repo/structure.sql (hunk 1 above)
Comment (tspenov) [reply priv/repo/structure.sql:1]: Seems`pg_dump` version 12.3 formats the sql. The only non compatible change from postgres 12 is `SET default_table_access_method = 'heap'` which I've removed from `structure.sql`.

Comment (IvanIvanoff) [reply priv/repo/structure.sql:1]: well, I should upgrade to postgres 12 or my next PR will have exactly the opposite changes :smiley:

**Pull request merged**